lib/gems/pending/util/mount/miq_s3_session.rb

```diff
@@ -0,0 +1,66 @@
+require 'util/mount/miq_generic_mount_session'
+require 'aws-sdk'
+
+class MiqS3Session < MiqGenericMountSession
+  def initialize(log_settings)
+    log_header = "MIQ(#{self.class.name}-initialize)"
+    logger.debug("#{log_header} initialize: log_settings are #{log_settings}")
+    super(log_settings)
+    logger.debug("#{log_header} initialize: @settings are #{@settings}")
+    raise "username is a required value!" if @settings[:username].nil?
+    raise "password is a required value!" if @settings[:password].nil?
+    raise "region is a required value!" if @settings[:region].nil?
+    @s3 = s3
+  end
+
+  def s3
+    @s3 ||= Aws::S3::Resource.new(:region => @settings[:region], :access_key_id => @settings[:username], :secret_access_key => @settings[:password])
+  end
+
+  def connect
+    @host       = URI(uri).host
+    @mount_path = URI(uri).path
+    super
+  end
+
+  def disconnect
+    return if @mnt_point.nil?
+    FileUtils.rm_rf(@mnt_point) if File.exist?(@mnt_point)
+  end
+
+  def uri_to_local_path(remote_file)
```

Comment:
Also I think adding tests for methods like this one and `uri_to_object_path` would make it much easier to understand what they were trying to do.